src/uct/tcp/tcp_iface.c

```diff
@@ -223,11 +223,15 @@ uct_tcp_iface_parse_virtual_dev(const struct dirent *entry, void *ctx)
 {
```

Comment:
There's code duplication in `uct_tcp_iface_parse_virtual_dev` and `ucs_netif_parse_bond_dev`.
Make it a single function and call it from both places.
The function can find and return the lowest device (stop when you can't find "lower_" prefix).

Reply:
Hi @shasson5, those two functions are different. the uct_tcp_iface_parse_virtual_dev is to find the lowest device(/sys/class/net/bond0.192/lower_bond0/lower_ens4f0np0). But ucs_netif_parse_bond_dev is just to find the first bond device in deep search loop. For example, it just find /sys/class/net/bond0.192/lower_bond0, not /sys/class/net/bond0.192/lower_bond0/lower_ens4f0np0. Thanks

Reply:
Hi @shasson5 I move the code into `ucs_topo_iface_get_lower_dev_path` and call it in both `uct_tcp_iface_parse_virtual_dev` and `ucs_netif_parse_bond_dev`. Please review. Thanks.